### PR TITLE
[plugins-commons] fix: disable inner-focus when accordion is selected

### PR DESCRIPTION
**Pull request by maimonav**
When multiple blocks are selected, and Accordion is one of them, disable inner-focus of plugin and wrap it with border to let the user acknowledge that the plugin has been selected.

### CODE REVIEW

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx [hunk 1]
```diff
@@ -274,6 +270,10 @@ const createBaseComponent = ({
       const { width: currentWidth, height: currentHeight } = componentData.config || {};
       const { width: initialWidth, height: initialHeight } = settings || {};
       const isEditorFocused = selection.getHasFocus();
+      const isOneBlockSelected = selection.getStartKey() === selection.getEndKey();
+      const isSelected = blockProps.isFocused && !isOneBlockSelected;
```
Comment (shaulgo) [inline packages/plugin-commons/web/src/Base/createBaseComponent.jsx:274]: When one block is selected, it's still selected? 
```suggestion
      const isSelected = blockProps.isFocused
```

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx [hunk 2]
```diff
@@ -274,6 +270,10 @@ const createBaseComponent = ({
...
+      blockProps.isFocused = blockProps.isFocused && isOneBlockSelected;
+
       const { isFocused } = blockProps;
       const isActive = isFocused && isEditorFocused;
```
Comment (shaulgo) [inline packages/plugin-commons/web/src/Base/createBaseComponent.jsx:278]: do we need this?
My logic says that 
isSelected = when it's selected
isFocused = when _only it's selected_ && _editor is focused_, should be enough.

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -39,8 +44,8 @@
 - `editor`
   - [#1706](https://github.com/wix-incubator/rich-content/pull/1706) pasting text with formatting from other editors (Google Docs, MS Word..)
 ### :bug: Bug Fix
-  - `editor-common`
-    - [#1707](https://github.com/wix-incubator/rich-content/pull/1707) fix checkbox focus-ring
+- `editor-common`
```
[the remark this answers is not in the document]
Comment (maimonav) [reply CHANGELOG.md:47]: I did it to fix the spacing in the changelog, do you want me to change it straight from master?

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx (hunk 1 above)
Comment (maimonav) [reply packages/plugin-commons/web/src/Base/createBaseComponent.jsx:274]: I see 2 use cases here:
1. the block is part of selection (multiple blocks selected), this case is only to support border on every plugin.
2. the block is focused (only the block is selected).

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx (hunk 2 above)
Comment (maimonav) [reply packages/plugin-commons/web/src/Base/createBaseComponent.jsx:278]: Check out the change that I've made, isActive is just isFocused.

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx (hunk 1 above)
Comment (maimonav) [reply packages/plugin-commons/web/src/Base/createBaseComponent.jsx:274]: And no, if there is one block selected and the block is focused - than the block is focused.

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx [hunk 4]
```diff
@@ -297,17 +296,16 @@ const createBaseComponent = ({
         classNameStrategies,
         className || '',
         {
-          [this.styles.hasFocus]: isActive && !pluginsWithoutBorderOnFocus.includes(type),
-          [theme.hasFocus]: isActive,
+          [this.styles.hasFocus]: (isFocused && !noPluginBorder) || isSelected,
```
Comment (shaulgo) [inline packages/plugin-commons/web/src/Base/createBaseComponent.jsx:299]:
```suggestion
          [this.styles.hasFocus]: isFocused ? !noPluginBorder : isSelected,
```

**Review by shaulgo — APPROVED** [review]

File: packages/plugin-commons/web/src/consts.ts [hunk 5]
```diff
@@ -35,12 +37,10 @@ export const FOOTER_BUTTON_ALIGNMENT = Object.freeze({
...
+export const pluginsWithoutBorderOnHover = [ACCORDION];
 
-export const pluginsWithoutBorderOnHover = [PLUGINS_TYPE_MAP.ACCORDION];
+export const pluginsWithoutBorderOnFocus = [ACCORDION];
 
-export const pluginsWithoutBorderOnFocus = [PLUGINS_TYPE_MAP.ACCORDION];
+export const pluginsWithoutPointerEventsOnFocus = [ACCORDION];
 
-export const pluginsWithoutPointerEventsOnFocus = [PLUGINS_TYPE_MAP.ACCORDION];
+export const pluginsWithoutInnerFocusWhenMultipleBlocksSelected = [ACCORDION];
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/plugin-commons/web/src/consts.ts]: done.

File: packages/plugin-commons/web/src/consts.ts [hunk 6]
```diff
@@ -35,12 +37,10 @@ export const FOOTER_BUTTON_ALIGNMENT = Object.freeze({
...
-const PLUGINS_TYPE_MAP = {
-  ACCORDION: 'wix-rich-content-plugin-accordion',
-};
+export const pluginsWithoutBorderOnHover = [ACCORDION];
 
-export const pluginsWithoutBorderOnHover = [PLUGINS_TYPE_MAP.ACCORDION];
+export const pluginsWithoutBorderOnFocus = [ACCORDION];
 
-export const pluginsWithoutBorderOnFocus = [PLUGINS_TYPE_MAP.ACCORDION];
+export const pluginsWithoutPointerEventsOnFocus = [ACCORDION];
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/plugin-commons/web/src/consts.ts]: done.

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx [hunk 7]
```diff
@@ -297,8 +310,10 @@ const createBaseComponent = ({
         classNameStrategies,
         className || '',
         {
-          [this.styles.hasFocus]: isActive && !pluginsWithoutBorderOnFocus.includes(type),
+          [this.styles.hasFocus]:
+            (isActive && !pluginsWithoutBorderOnFocus.includes(type)) || shouldDisableInnerFocus,
           [theme.hasFocus]: isActive,
+          [this.styles.hideSelection]: !isActive,
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/plugin-commons/web/src/Base/createBaseComponent.jsx]: done.

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx [hunk 8]
```diff
@@ -274,6 +276,15 @@ const createBaseComponent = ({
       const { width: currentWidth, height: currentHeight } = componentData.config || {};
       const { width: initialWidth, height: initialHeight } = settings || {};
       const isEditorFocused = selection.getHasFocus();
+      const shouldDisableInnerFocus =
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/plugin-commons/web/src/Base/createBaseComponent.jsx]: done.

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx [hunk 9]
```diff
@@ -297,17 +296,16 @@ const createBaseComponent = ({
         classNameStrategies,
         className || '',
         {
-          [this.styles.hasFocus]: isActive && !pluginsWithoutBorderOnFocus.includes(type),
-          [theme.hasFocus]: isActive,
+          [this.styles.hasFocus]: isFocused ? !noPluginBorder : isSelected,
+          [theme.hasFocus]: isFocused,
```
Comment (shaulgo) [inline packages/plugin-commons/web/src/Base/createBaseComponent.jsx:300]:
```suggestion
          [theme.hasFocus]: isFocused ? !noPluginBorder : isSelected,
```

File: packages/plugin-commons/web/src/Base/createBaseComponent.jsx [hunk 10]
```diff
@@ -288,7 +287,7 @@ const createBaseComponent = ({
         this.styles.pluginContainer,
         theme.pluginContainer,
         theme.pluginContainerWrapper,
-        pluginsWithoutBorderOnHover.includes(type) && this.styles.noBorderOnHover,
+        !isSelected && noPluginBorder && this.styles.noBorderOnHover,
```
Comment (shaulgo) [inline packages/plugin-commons/web/src/Base/createBaseComponent.jsx:290]: !(isSelected && !isFocused && noPluginBorder) && this.styles.noBorder

Comment (maimonav) [reply packages/plugin-commons/web/src/Base/createBaseComponent.jsx]: I changed it a bit, since this was not the solution.

**Pull request merged**